src/wui/savegameloader.cc

```diff
@@ -0,0 +1,221 @@
+			gamedata.savedonstring =
+			   /** TRANSLATORS: Display date for choosing a savegame/replay. Placeholders are:
+			                                                    hour:minute. This is part of a list.
+			    */
+			   (boost::format(_("saved today at %1%:%2%")) % savedate->tm_hour % minute).str();
+		} else if ((savedate->tm_year == current_year && savedate->tm_mon == current_month &&
+		            savedate->tm_mday == current_day - 1) ||
+		           (savedate->tm_year == current_year - 1 && savedate->tm_mon == 11 &&
+		            current_month == 0 && savedate->tm_mday == 31 &&
+		            current_day == 1)) {  // Yesterday
+			// Adding the 0 padding in a separate statement so translators won't have to deal
+			// with it
+			const std::string minute = (boost::format("%02u") % savedate->tm_min).str();
+
+			gamedata.savedatestring =
+			   /** TRANSLATORS: Display date for choosing a savegame/replay. Placeholders are:
+			                                                    hour:minute */
+			   (boost::format(_("Yesterday, %1%:%2%")) % savedate->tm_hour % minute).str();
+			gamedata.savedonstring =
+			   /** TRANSLATORS: Display date for choosing a savegame/replay. Placeholders are:
+			                                                    hour:minute. This is part of a list.
+			    */
+			   (boost::format(_("saved yesterday at %1%:%2%")) % savedate->tm_hour % minute).str();
+		} else {  // Older
+			gamedata.savedatestring =
+			   /** TRANSLATORS: Display date for choosing a savegame/replay. Placeholders are:
+			                                                    month day, year */
+			   (boost::format(_("%1% %2%, %3%")) % localize_month(savedate->tm_mon) %
+			    savedate->tm_mday % (1900 + savedate->tm_year))
+			      .str();
+			gamedata.savedonstring =
+			   /** TRANSLATORS: Display date for choosing a savegame/replay. Placeholders are:
+			                                                    month (short name) day (number),
+			      year (number). This is part of a list. */
+			   (boost::format(_("saved on %1% %2%, %3%")) % savedate->tm_mday %
+			    localize_month(savedate->tm_mon) % (1900 + savedate->tm_year))
+			      .str();
+		}
+	}
+}
+
+void SavegameLoader::add_sub_dir(const std::string& gamefilename,
+                                 std::vector<SavegameData>& loaded_games) const {
+	// Add subdirectory to the list
+	const char* fs_filename = FileSystem::fs_filename(gamefilename.c_str());
+	if (!strcmp(fs_filename, ".") || !strcmp(fs_filename, "..")) {
+		return;
+	}
+	loaded_games.push_back(SavegameData::create_sub_dir(gamefilename));
+}
+
+ReplayLoader::ReplayLoader(Widelands::Game& game) : SavegameLoader(game) {
+}
+
+bool ReplayLoader::is_valid_gametype(const SavegameData&) const {
+	return true;  // why?? what is the purpose of GameController::GameType::kReplay?
+	              // return gamedata.is_replay(); <-- should be this, right?!
```

Comment:
Right :)

Reply:
yeah... no. When I do this, at least in my tests no games/replays whatsoever are shown anymore. The original code always returns true so "normal" savegames are presented after clicking "Watch replay"-Button. This is why I asked how the system saves a game so it "becomes" a replay...

Reply:
I think I remember now that replays were tagged as one of the multiplayer game types. That might be a bug - we never dug into that code.

Reply:
Can you add `TODO(jmoerschbach ):` to that comment? I'd say fixing this problem is out of scope for this branch.

Reply:
I added a TODO